dns.go

```diff
@@ -152,7 +155,10 @@ func (api *API) CreateDNSRecord(ctx context.Context, rc *ResourceContainer, para
 	return recordResp, nil
 }
 
-// ListDNSRecords returns a slice of DNS records for the given zone identifier.
+// ListDNSRecords returns a slice of DNS records for the given zone
+// identifier. If params doesn't include any pagination (ResultInfo)
+// options, auto pagination is performed with the default page size
+// of 100 records per request.
```

Comment:
given this is a library convention (and default) we can omit this

```suggestion
// ListDNSRecords returns a slice of DNS records for the given zone
// identifier. 
```